migrate_pim/upgrade_from_32_to_40.rst

```diff
@@ -0,0 +1,218 @@
+You can check the VirtualHost configuration for 4.0: system_requirements/system_configuration.rst.inc.
+
+
+Prepare your project
+********************
+
+.. warning::
+
+    All the following commands need to be run from the root of your 3.2 PIM
+
+.. code:: bash
+
+    $ rm -rf var/cache/
+    $ tar --strip-component 1 -xvzf /path/to/pim-enterprise-standard-v4.0.0.tar.gz  pim-enterprise-standard/composer.json
+
+Load your PIM Enterprise 4.0 dependencies
+*****************************************
+
+.. code:: bash
+
+    $ composer update
+
+
+.. note::
+
+    You may need to increase the memory provided to `composer`, as this step can be very memory consuming:
+
+    .. code:: bash
+
+        $ php -d "memory_limit=4G" /path/to/composer update
+
+Let Akeneo PIM 4.0 continue the preparation for you
+***************************************************
+
+.. code:: bash
+
+    $ vendor/akeneo/pim-enterprise-dev/std-build/migration/prepare_32_to_40.sh
+
+
+.. note::
+
+    This script will update the current filesystem layout to match Symfony 4 structure.
+    See http://fabien.potencier.org/symfony4-directory-structure.html for more details.
+
+
+MySQL and ES Credentials Access
+*******************************
+
+Akeneo PaaS
+-----------
+
+If you are using an Akeneo PaaS env, the credentials for MySQL and ES are already available as environment variables.
+These environment variables will be directly available to your Akeneo PIM.
+
+Local or on-premise environment
+-------------------------------
+
+You can make the variables content available to your program or set them in a .env.local file:
+
+.. code::
+
+    APP_DATABASE_HOST=mysql-host
+    APP_DATABASE_PORT=3306
+    APP_DATABASE_NAME=akeneo_pim_db_name
+    APP_DATABASE_USER=akeneo_pim_user
+    APP_DATABASE_PASSWORD=akeneo_pim_password
+    APP_INDEX_HOSTS=elasticsearch-host:9200
+
+
+Make sure your environment is ready to be migrated
+**************************************************
+
+.. code:: bash
+
+    $ bin/console pim:installer:check-requirements
+
+
+If this command detects something not working or not properly configured,
+please fix the problem before continuing.
+
+Prepare the front
+*****************
+
+.. code:: bash
+
+    $ bin/console pim:installer:assets --symlink --clean
+    $ yarnpkg install
+    $ yarnpkg run webpack
+
+Migrate your data
+*****************
+
+.. code:: bash
+
+    $ bin/console doctrine:migration:migrate
+
+
+.. note::
+
+    You may receive the following warnings:
+
+        WARNING! You have X previously executed migrations in the database that are not registered migrations.
+
+    This can be safely ignored as this only means that your database is up to date, but without finding the corresponding
+    migration files.
+
+    Another message could be `Migration _X_Y_ZZZZ was executed but did not result in any SQL statements`.
+
+    This makes sense for some migration that only touches the Elasticsearch index or don't apply because no data linked
```

Comment:
Actually, it's because we use the connection of the container. But probably useless information.